### ISSUE TITLE
java.lang.NullPointerException

### ISSUE THREAD

**Issue by ildfnkd** [hannesa2/paho.mqtt.android#732]
# MQTT Rec: ANDROID-ELE-AL00-cd7aff97e9ba3972(2855)

java.lang.NullPointerException

Attempt to invoke virtual method 'long org.eclipse.paho.client.mqttv3.internal.ClientState.getKeepAlive()' on a null object reference
org.eclipse.paho.client.mqttv3.internal.ClientComms.getKeepAlive(ClientComms.java:656)
info.mqtt.android.service.ping.AlarmPingSender.start(AlarmPingSender.kt:28)
org.eclipse.paho.client.mqttv3.internal.ClientState.connected(ClientState.java:1214)
org.eclipse.paho.client.mqttv3.internal.ClientState.notifyReceivedAck(ClientState.java:1050)
org.eclipse.paho.client.mqttv3.internal.CommsReceiver.run(CommsReceiver.java:151)
java.lang.Thread.run(Thread.java:932)

**Comment by hannesa2** [comment]
too less information

**Comment by ildfnkd** [comment]
@hannesa2 When I use it for the first time, I occasionally have a problem with the null pointer, or is there a problem with initialization?

**Comment by hannesa2** [comment]
What's about version name and a running sample ?

**Comment by FJEagle** [comment]
Might same as #433

**Comment by HMeckler** [comment]
I am having this same issue on android devices OS 30 using: implementation 'com.github.hannesa2:paho.mqtt.android:4.2.4'

**Comment by hannesa2** [comment]
> I am having this same issue on android devices OS 30 using: implementation 'com.github.hannesa2:paho.mqtt.android:4.2.4'

Please give `4.4` a try

**Comment by mohammaddvlpr** [comment]
this happens for me too and i dont have any scenario. 

It may be handled by putting the block inside try catch and return default value (zero) if any exception is thrown(if it's ok, i can create a pull request to just silent it this way)

here's the code that throws exception inside ClientComms class:

`public long getKeepAlive() {
		return this.clientState.getKeepAlive();
	}`

the lib's version is 3.6.6.

**Comment by hannesa2** [comment]
Yes, to give me a chance to analyze, please make a pull request which shows me the issue

**Comment by mohammaddvlpr** [comment]
Unfortunately the exception is thrown in eclipse paho lib in ClientComms class, is there any way to fix that? changing paho lib is risky i think

**Comment by hannesa2** [comment]
> changing paho lib is risky i think

I see it not risky, I see it just as some additional work.

As these guys from https://github.com/eclipse-paho/paho.mqtt.java moving super slow (or event don't want to move) and decline my offer (years ago) to be a maintainer there, I made [a fork with CI](https://github.com/hannesa2/paho.mqtt.java/releases) and use it here https://github.com/hannesa2/paho.mqtt.android/pull/927

When you know what to do, to avoid this issue in my fork, please make a pull request there and I'll merge, create a release and use it here

**Comment by hannesa2** [comment]
@ildfnkd 
@FJEagle 
@mohammaddvlpr 
@HMeckler 
Please use `implementation("com.github.hannesa2:paho.mqtt.android:4.6-beta1")` and give me feedback.
I will not merge https://github.com/hannesa2/paho.mqtt.android/pull/927 without feedback.

**Comment by mohammaddvlpr** [comment]
I'll do it in the following days, thanks for your support.

@hannesa2

**Comment by mohammaddvlpr** [comment]
Can you make a release based on version 3.6.6 @hannesa2 ? currently i'm using this version and has not migrated to the latest version.

**Comment by hannesa2** [comment]
@mohammaddvlpr It's too much work https://github.com/hannesa2/paho.mqtt.android/pull/930. Please make a pull request which solves the issues for old 3.x branch, the I can make a release.
Btw, I would recommend to move to latest 4.x

**Comment by mohammaddvlpr** [comment]
I only changed the version of eclipse paho to your newly forked version of paho then created a PR [here](https://github.com/hannesa2/paho.mqtt.android/pull/931), is that ok? @hannesa2

**Comment by hannesa2** [comment]
Thank you, as I merged too fast, I fixed it now https://github.com/hannesa2/paho.mqtt.android/pull/933
and you can test it with 
`implementation("com.github.hannesa2:paho.mqtt.android:3.7-beta2")`
and give me feedback. If it's positive, then I'll create `3.7`

**Comment by mohammaddvlpr** [comment]
Thank you very much @hannesa2
